### PR TITLE
TINY-10453: Remove `title` attribute for menu items, add `data-mce-label` for toolbar and dialog buttons

### PR DESCRIPTION
**Pull request by shanmen-tiny**
Related Ticket: TINY-10453

Description of Changes:
* Attribute changes to buttons are made in this commit: https://github.com/tinymce/tinymce/commit/37c5adf55ecd972e0bc975ebc5f0c85b6843499f
	* Added `data-mce-label` for toolbar and dialog buttons
	* Removed `title` attribute for buttons that have visible label, see notion link for more detail
	* `renderCommonDropdown` now takes `tooltip` and `ariaLabel` properties, `listBox` requires the `aria-label` attribute but not the tooltip
* Tests changes for attribute changes (removal of title attribute and `data-mce-label` are made in this commit: https://github.com/tinymce/tinymce/commit/49ee66a1af030f33d9e4e54ebd5bf6e8082cfb41
* Tootip tests for different buttons: https://github.com/tinymce/tinymce/commit/83c10ed0fe42a4382e135cbe1aeacf88c5aedb9e
* Remove the `title` attribute from directory and tree item: https://github.com/tinymce/tinymce/pull/9312/commits/98e9eb383072dee91c8a5633342291a828792c46
* Removed the `title` attribute for `ViewButton` as it already has a visible label. Also removed title attribute when dynamically updating the `aria-label` for `MenuItem`  when state changes . https://github.com/tinymce/tinymce/pull/9312/commits/77a0f1a94410d1b72cfc227b01f922597c72d38c

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/SizeInput.ts
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/SizeInput.ts:37]: Probably worth having a label variable like `const label = spec.label.getOr('Constrain proportions')`; to avoid the duplication

**Review by ltrouton — APPROVED** [review]

**Comment by spocke** [comment]
Could the property be called `data-mce-title` or `data-mce-label` since `btn` kind of makes it specific to just buttons a custom label could be on other things as well like checkboxes, input fields and thing that can have focus.

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeSelect.ts
Comment (MitchC1999) [inline modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeSelect.ts:182]: Seeing as no bespoke buttons can be made via public API, and we are now always passing in the btnName, should the btnName not be optional?

**Comment by shanmen-tiny** [comment]
> Could the property be called `data-mce-title` or `data-mce-label` since `btn` kind of makes it specific to just buttons a custom label could be on other things as well like checkboxes, input fields and thing that can have focus.

Yeah not a bad idea, I'm inclined towards `data-mce-label` so I've just renamed that. Hopefully, this attribute will also be used in other places in the future.

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/SizeInput.ts
Comment (shanmen-tiny) [reply modules/tinymce/src/themes/silver/main/ts/ui/dialog/SizeInput.ts]: Fixed 👍  https://github.com/tinymce/tinymce/pull/9312/commits/61bd86bad49010bf1144e64bf6b967b19e5ed11b

**Comment by TheSpyder** [comment]
`label` makes it sound like a text label, though. This is the codename used to register the button in code.

Perhaps `data-mce-name`?

File: modules/tinymce/src/plugins/table/test/ts/module/test/TableModifiersTestUtils.ts
Comment (TheSpyder) [inline modules/tinymce/src/plugins/table/test/ts/module/test/TableModifiersTestUtils.ts:116]: This is the sort of reason why I wanted to apply the `data-mce-name` concept everywhere and never use `aria-label` in tests 🙃 it would make the tests so much easier and more consistent

But there isn't scope for that now.

**Review by TheSpyder — APPROVED** [review]

**Comment by shanmen-tiny** [comment]
> `label` makes it sound like a text label, though. This is the codename used to register the button in code.
> 
> Perhaps `data-mce-name`?
@TheSpyder Renamed to `data-mce-name`. Tagging @spocke, tick if you're happy with that as well. 👍

**Comment by spocke** [comment]
> `label` makes it sound like a text label, though. This is the codename used to register the button in code.
> 
> Perhaps `data-mce-name`?

Right, I thought it was just a renamed title to not show it. Looks like it's a button name of what the control id is so yeah then name is more correct. But I wonder if this will now be abused to custom style individual buttons if all controls have them but maybe that is fine. Technically one could have done this with bth-child or matching the title before but less elegant and probably more error prone.

Also this PR needs a similar PR for the monorepo before this one can be merged, since now when we constant pull in changes in that from develop we need to fix it all at once.

**Comment by TheSpyder** [comment]
> I wonder if this will now be abused to custom style individual buttons if all controls have them but maybe that is fine

It's a risk, for sure, but it's better than using the english text to locate buttons in tests.

**Comment by spocke** [comment]
> > I wonder if this will now be abused to custom style individual buttons if all controls have them but maybe that is fine
> 
> It's a risk, for sure, but IMO it's better than using the english text to locate buttons in tests.

Yes people who in the past raced concern about this has left. I remember there where concerns of adding extra properties just used for tests etc. Looks nice from a dev perspective now we can make mcagar simpler and reduce duplicate code of querying for buttons using selectors in a million places.

**Pull request merged**
